### ISSUE TITLE
External API usage

### ISSUE THREAD

**Issue by realgooseman** [jdejaegh/irm-kmi-ha#61]
Hi @jdejaegh, 

Thanks for this project, it has been made by reverse engineering the mobile app right ?

I would like to know if it's possible to implement that API in other FOSS weather apps.
If so, how could I inform the devs, any API usage docs outside Home Assistant ?

**Describe the solution you'd like**
Some documentation on how to use the API to add weather data on other weather apps (like **[Breezy Weather](https://github.com/breezy-weather/breezy-weather)** or **[FluentWeather](https://github.com/Gabboxl/FluentWeather)**).

**Comment by jdejaegh** [comment]
Hi,

> Thanks for this project, it has been made by reverse engineering the mobile app right ?

Exactly!

I could indeed describe the API in more details on a wiki page.

**Comment by realgooseman** [comment]
Great, thanks for this project.

I suppose there is no official API for IRM/KMI right ? Or maybe it's paid. I don't see any API docs on their website either. Only a link to download data chunks via Open Data.

**Comment by jdejaegh** [comment]
I did not find any API documentation either. That's why I ended up reverse engineering the Android application.

The open data contains raw measures and outputs from their models, it is not really usable unless you are a meteorologist... Interesting datasets (forecast by municipalities and similar) are not public. That's what I found last time I checked.

**Comment by realgooseman** [comment]
Yeah, I came to the same conclusion. I'll try contacting them after the holidays maybe they're working on it or will allow API access for FOSS non-profit projects.

Do you think your reverse engineered API is stable and will be future proof or can it break at any time and will need to be re-engineered if they change something on the app/server side ?

**Comment by jdejaegh** [comment]
I made this integration such that it uses `github.com/jdejaegh/irm-kmi-ha` as User-Agent header so they can see that not only their app is using the API endpoint.  I don't know if they monitor that.

Regarding stability, I can only tell from observation: the integration has been there since one year, and there have been no breaking changes since then.  If they add fields to the JSON responses in the future, it should not break the integration.  If they modify or remove fields, it will break for sure. 

At the moment, I don't have tests to verify that their API is still returning consistent values.  All the tests assume that the responses fall within the range of responses I observed in the past.

**Comment by jdejaegh** [comment]
A first version is available in the wiki: https://github.com/jdejaegh/irm-kmi-ha/wiki/API-documentation

It is not yet complete but already provides an overview of how the API works.  I plan to describe each method when I have time.  The biggest is `getForecasts` as it returns the full weather data.

I'll close this issue once the page is done.

**Comment by jdejaegh** [comment]
It looks good to me: https://github.com/jdejaegh/irm-kmi-ha/wiki/API-documentation

If you discover more methods or want to update parts of it, feel free to open an issue.
